src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,109 +31,128 @@
+
+            host_task<detail::kernel_name<mklcpu_desc_t>>(cgh, [=]() {
+                std::int64_t local_err = DFTI_MEMORY_ERROR;
+                if (config_values.dimensions.size() == 1)
+                    local_err = DftiCreateDescriptor(handle_obj.get_pointer(), mklcpu_prec,
+                                                     mklcpu_dom, 1, config_values.dimensions[0]);
+                else
+                    local_err = DftiCreateDescriptor(handle_obj.get_pointer(), mklcpu_prec,
+                                                     mklcpu_dom, config_values.dimensions.size(),
+                                                     config_values.dimensions.data());
+
+                set_value(*handle_obj.get_pointer(), config_values);
+                if (local_err != DFTI_NO_ERROR) {
+                    throw oneapi::mkl::exception("dft", "commit", "DftiCreateDescriptor failed");
+                }
+
+                local_err = DftiCommitDescriptor(*handle_obj.get_pointer());
+                *status_obj.get_pointer() = local_err;
+            });
+        });
+
+        status = status_buffer.template get_access<sycl::access::mode::read>()[0];
         if (status != DFTI_NO_ERROR) {
             throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
                                          "DftiCommitDescriptor failed");
         }
+
+        device_handle = handle_buffer.template get_access<sycl::access::mode::read>()[0];
     }
 
-    virtual void* get_handle() noexcept override {
-        return handle;
+    void* get_handle() noexcept override {
+        return reinterpret_cast<void*>(device_handle);
     }
 
     virtual ~commit_derived_impl() override {
-        DftiFreeDescriptor((DFTI_DESCRIPTOR_HANDLE*)&handle);
+        DftiFreeDescriptor(&device_handle);
     }
 
 private:
-    DFTI_DESCRIPTOR_HANDLE handle = nullptr;
-
-    constexpr DFTI_CONFIG_VALUE get_domain(domain d) {
-        if (d == domain::COMPLEX) {
-            return DFTI_COMPLEX;
-        }
-        else {
-            return DFTI_REAL;
-        }
-    }
-
-    constexpr DFTI_CONFIG_VALUE get_precision(precision p) {
-        if (p == precision::SINGLE) {
-            return DFTI_SINGLE;
-        }
-        else {
-            return DFTI_DOUBLE;
-        }
-    }
+    mklcpu_desc_t device_handle = nullptr;
 
     template <typename... Args>
-    void set_value_item(DFTI_DESCRIPTOR_HANDLE hand, enum DFTI_CONFIG_PARAM name, Args... args) {
-        if (auto ret = DftiSetValue(hand, name, args...); ret != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception(
-                "dft/backends/mklcpu", "set_value_item",
-                "name: " + std::to_string(name) + " error: " + std::to_string(ret));
+    void set_value_item(mklcpu_desc_t hand, enum DFTI_CONFIG_PARAM name, Args... args) {
+        DFT_ERROR value_err = DFTI_NOTSET;
+        value_err = DftiSetValue(hand, name, args...);
```

Comment:
```suggestion
       DFT_ERROR value_err = DftiSetValue(hand, name, args...);
```
You missed this bit of the suggestion

Reply:
https://github.com/oneapi-src/oneMKL/pull/288#discussion_r1136933569